pkg/podexecutor/staticpod.go

```diff
@@ -348,6 +351,8 @@ func (s *StaticPodConfig) ControllerManager(apiReady <-chan struct{}, args []str
 		}
 		args = append(extraArgs, args...)
 	}
+	args = append(args, permitPortSharingFlag)
```

Comment:
nit: This needs to be done in the reverse order (appending args to the args added here) to ensure that args added by the user come last and supersede any of our defaults.
```suggestion
	args = append(permitPortSharingFlag, args...)
```